chore: add Js string methods transcript (#58897)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-common-string-methods/67326c15b3b2f0c5827927cc.md

# --description--

Watch the video or read the transcript and answer the questions below.

# --transcript--

How can you extract a substring from a string?

When working with strings in JavaScript, you often need to extract a portion or substring from a larger string.

For example, you may want to extract part of a word, a specific character sequence, or just a fragment of a sentence.

JavaScript provides several methods for this task, one of the most commonly used being the `slice()` method.

The `slice()` method allows you to extract a portion of a string and returns a new string, without modifying the original string. It takes two parameters: the starting index and the optional ending index.

Here's the basic syntax:

```js
string.slice(startIndex, endIndex);
```

`startIndex` is the position where the extraction starts. `endIndex` is where the extraction ends. If not provided, `slice()` extracts until the end of the string.

> Let’s look at a simple example of extracting part of a string:

Reviewer: ```suggestion
Let's look at a simple example of extracting part of a string:
```

```js
let message = "Hello, world!";
let greeting = message.slice(0, 5);

console.log(greeting);  // Output: Hello
```

> In this example, `slice(0, 5)` extracts characters starting from index `0` up to but not including index `5`. As a result, the word `"Hello"` is extracted.

Reviewer: ```suggestion
In this example, `slice(0, 5)` extracts characters starting from index `0` up to but not including index `5`. As a result, the word `Hello` is extracted.
```

If you omit the second parameter, `slice()` will extract everything from the start index to the end of the string:

```js
let message = "Hello, world!";
let world = message.slice(7);

console.log(world);  // Output: world!
```

> Here, `slice(7)` extracts the string from index `7` to the end of the string, resulting in `"world!"`.

Reviewer: ```suggestion
Here, `slice(7)` extracts the string from index `7` to the end of the string, resulting in `world!`.
```

You can also use negative numbers as indexes. When you use a negative number, it counts backward from the end of the string:

```js
let message = "JavaScript is fun!";
let lastWord = message.slice(-4);

console.log(lastWord);  // Output: fun!
```

> In this case, `slice(-4)` extracts the last four characters from the string, giving us `"fun!"`.

Reviewer: ```suggestion
In this case, `slice(-4)` extracts the last four characters from the string, giving us `fun!`.
```

> Let’s say you want to extract a section from the middle of a string. You can provide both the starting and ending indexes to precisely control which part of the string you want:

Reviewer: ```suggestion
Let's say you want to extract a section from the middle of a string. You can provide both the starting and ending indexes to precisely control which part of the string you want:
```

```js
let message = "I love JavaScript!";
let language = message.slice(7, 17);

console.log(language);  // Output: JavaScript
```

> Here, `slice(7, 17)` extracts the substring starting at index 7 and ending right before index `17`, which is the word `"JavaScript"`.

Reviewer: ```suggestion
Here, `slice(7, 17)` extracts the substring starting at index 7 and ending right before index `17`, which is the word `JavaScript`.
```
